cmd/operator-sdk/generate/csv.go

```diff
@@ -320,3 +328,33 @@ func copyCustomResourceDefinitions(fromDir, toDir string) error {
 
 	return nil
 }
+
+func getDefaults() *csvCmd {
+	c := &csvCmd{}
+
+	if kbutil.IsConfigExist() {
+		cfg, err := kbutil.ReadConfig()
+		if err != nil {
+			log.Fatal(err)
+		}
+		c.operatorName = filepath.Base(cfg.Repo)
+		c.deployDir = "config"
+		c.crdDir = filepath.Join("config", "bases")
```

Comment:
```suggestion
		c.crdDir = filepath.Join("config", "crd" "bases")
```

Also with `c.deployDir = "config"` the CSV generator's `updateCSVFromManifests()` will walk the entire directory and pick up extraneous `Deployment` and `Role/ClusterRole` manifests.
For e.g the auth proxy and webhook patches will each result in a separate deployment strategy:
https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v2/config/default/manager_auth_proxy_patch.yaml
https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v2/config/default/manager_webhook_patch.yaml

And we can't just parse the base operator manifest at `c.deployDir = "config/manager"` since then we miss out on the deployment patches to the spec. Also we're using the `--deploy-dir` flag to also parse RBAC manifests and CR samples so we can't point to a single path like `config/manager`.

For kubebuilder alignment I'm rewriting `updateCSVFromManifests()` to run `kustomize build config/default` to parse the Deployment, Role/ClusterRole and CR sample manifests.

So in Kubebuilder's layout the `--deploy-dir` flag is going to be ignored.
Or to be more clear it's going to point to the root kustomize directory that we expect to `kustomize build` and parse for operator manifests. The default being `config/default` that users could then configure if they have a different set of kustomize operator manfiests to generate CSVs from e.g `config/production`.

I don't know if we want to keep the current behavior of parsing complete manifests from `--deploy-dir` in Kubebuilder's layout since we're always expecting kustomize manifests. I don't see that as being a use case when using Kubebuilder's layout but if it is we can always add that back via another flag.

Reply:
It may be easier to rewrite `generate csv` to receive input from stdin so it can be run with make:
```make
generate-olm:
    kustomize build config/default | operator-sdk generate csv —output-dir config/

deploy-olm:
    kustomize build config/default | operator-sdk generate csv —output-dir - | kubectl apply -f -
```
similar to the current `manifests`/`deploy` targets. Calling kustomize then the generator using make would align with how kubebuilder does things. Thoughts?